src/app/[locale]/(routes)/loans/loan-form.tsx

```diff
@@ -0,0 +1,143 @@
+import { createLoanAction } from '@/app/_action';
+import { TLoan } from '@/models/loan';
+
+const sampleData: TLoan = {
+	borrowerAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0A',
+	collateralAmount: '100',
+	collateralToken: '0x6B175474E89094C44Da98b954EedeAC495271d0B',
+	loanAmount: '100',
+	loanToken: '0x6B175474E89094C44Da98b954EedeAC495271d0C',
+	loanRepayDeadline: new Date(2023 - 12 - 31),
+	initialThreshold: '0.5',
+	interestRate: '0.1',
+	liquidationThreshold: '0.8',
+	lenderAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0D',
+	loanRequestDeadline: new Date(2023 - 12 - 31),
+	managerNFT: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
+	managerNFTVersion: 'v1',
+};
+
+const emptyData: TLoan = {
+	borrowerAddress: '',
+	collateralAmount: '',
+	collateralToken: '',
+	loanAmount: '',
+	loanToken: '',
+	loanRepayDeadline: new Date(),
+	initialThreshold: '',
+	interestRate: '',
+	liquidationThreshold: '',
+	lenderAddress: '',
+	loanRequestDeadline: new Date(),
+	managerNFT: '',
+	managerNFTVersion: '',
+};
+
+export default function LoanForm() {
+	async function action(data: FormData) {
+		'use server';
+
+		let _data: TLoan = emptyData;
+
+		// loop through FormData and assign values to _data object which will be sent to server
+		data.forEach((value, key) => (_data = { ..._data, [key]: value }));
+
+		// Invoke server action to add new loan
+		await createLoanAction({ params: _data, path: '/loans' });
+	}
+
+	return (
+		<form
+			action={action}
+			key={Math.random()}
+			className="grid sm:grid-cols-2 md:grid-cols-3 gap-4 place-items-center my-16"
+		>
+			<input
+				type="text"
+				name="collateralAmount"
+				placeholder="Collateral Amount"
+				className="border rounded px-2 py-1 flex-1"
+			/>
```

Comment:
I want a custom Input that wrap the style, so is the same style through the application

Reply:
done